adr/0007-package-variables-files-revisit.md

```diff
@@ -0,0 +1,217 @@
+    - *Cons*:
+      - Could make things unweildy in large packages by requiring the author to specify noPrompt
+    - [Diagram](../.images/0007-package-variables-files-revisit/NoPrompt%20Behavior.drawio.svg)
+
+&nbsp;
+
+5. Dissallow `default: nil` and `prompt: false` to be used together on package create
+    - *Pros*:
+      - Moves potential errors from the package deployer to the package creator
+      - Eliminates a potentially confusing case from existing
+    - *Cons*:
+      - Depends on implementing a `default` pointer
+    - [Diagram](../.images/0007-package-variables-files-revisit/Error%20on%20Package%20Create%20Behavior.drawio.svg)
+
+&nbsp;
+
+## Decision
+
+It was decided to allow the `default` key to be nillable ([Default #3](#default)), to change `prompt` to `noPrompt` ([Prompt #4](#prompt)), and to dissallow `noPrompt` and `default: nil` being combined at create time ([Prompt #5](#prompt)).  This would result in the following workflow:
+
+![Decided Behavior](../.images/0007-package-variables-files-revisit/Decided%20Behavior.drawio.svg)
+
+```yaml
+variables:
+  #  This case is now invalid and errors on package create
+  #  - Package variables cannot use noPrompt: true without setting a default
+  #
+  #- name: "NOTHING"
+  #  noPrompt: true
+  #  default: nil  <- decided change
+  - name: "NO_PROMPT"
+    default: "no-prompt"
+    noPrompt: true
+  - name: "NO_DEFAULT"
+    # noPrompt: false <- decided change
+    # default: nil    <- decided change
+  - name: "DEFAULT"
+    default: "default"
+    # noPrompt: false <- decided change
+```
+
+```shell
+zarf package deploy package.zst --confirm --set NO_PROMPT=world --set NO_DEFAULT=today --set DEFAULT=tommorow
+# This would continue to work, setting NO_PROMPT to "world", NO_DEFAULT to "today" and DEFAULT to "tomorrow"
+
+zarf package deploy package.zst --confirm --set NO_DEFAULT=today
+# This would continue to work, setting NO_PROMPT to "no-prompt", NO_DEFAULT to "today" and DEFAULT to "default"
+
+zarf package deploy package.zst --confirm
+# This would not work, erroring out with: "Variable NO_DEFAULT does not have a default, when using `--confirm` you must specify it with `--set` or `--set-file`."
+
+
+zarf package deploy package.zst
+# This would continue to work, setting NO_PROMPT to "no-prompt" and prompting for NO_DEFAULT and DEFAULT
+```
+
+## Consequences
+
+This would require the following breaking changes:
+
+- Any variables that do not currently have a `default` key specified have a new error state to handle on `--confirm`
+- `noPrompt` (formerly `prompt: false`) must have a `default` key set in order to use that key.
+- Package authors must flip their `prompt` logic to `noPrompt`.
+
+This choice has the following benefits:
+
+- Prompting is now the default which helps our interactive persona
+- Errors asking the user to specify command flags only happen when they are already using the `--conrifm` flag
+- Implicit `""` defaults no longer are possible
```

Comment:
For clarity I would add to this line that explicit `""` is still able to be used if that is the author/user's desire

Reply:
Added